core/services/keystore/eth.go

```diff
@@ -370,50 +429,68 @@ func (ks *eth) GetStatesForChain(chainID *big.Int) (states []ethkey.State, err e
-		if s.EVMChainID.Equal(utils.NewBig(chainID)) {
-			states = append(states, *s)
-		}
+	for _, s := range ks.keyStates.ChainIDKeyID[chainID.String()] {
+		states = append(states, *s)
 	}
+	sort.Slice(states, func(i, j int) bool { return states[i].KeyID() < states[j].KeyID() })
 	return
 }
 
-func (ks *eth) GetV1KeysAsV2(f DefaultEVMChainIDFunc) (keys []ethkey.KeyV2, states []ethkey.State, _ error) {
+func (ks *eth) getV1KeysAsV2() (keys []ethkey.KeyV2, nonces []int64, fundings []bool, _ error) {
 	v1Keys, err := ks.orm.GetEncryptedV1EthKeys()
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to get encrypted v1 eth keys")
+		return nil, nil, nil, errors.Wrap(err, "failed to get encrypted v1 eth keys")
 	}
 	if len(v1Keys) == 0 {
-		return keys, states, nil
-	}
-	chainID, err := f()
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, `%d legacy eth keys detected, but no default EVM chain ID was specified
-
-PLEASE READ THIS ADDITIONAL INFO
-
-If you are running Chainlink with EVM_ENABLED=false and don't care about EVM keys at all, you can run the following SQL to remove any lingering eth keys that may have been autogenerated by an older version of Chainlink, and boot the node again:
-
-pqsl> TRUNCATE keys;
-
-WARNING: This will PERMANENTLY AND IRRECOVERABLY delete any legacy eth keys, so please be absolutely sure this is what you want before you run this. Consider taking a database backup first`, len(v1Keys))
+		return nil, nil, nil, nil
 	}
 	for _, keyV1 := range v1Keys {
 		dKey, err := keystore.DecryptKey(keyV1.JSON, ks.password)
 		if err != nil {
-			return nil, nil, errors.Wrapf(err, "could not decrypt eth key %s", keyV1.Address.Hex())
+			return nil, nil, nil, errors.Wrapf(err, "could not decrypt eth key %s", keyV1.Address.Hex())
 		}
 		keyV2 := ethkey.FromPrivateKey(dKey.PrivateKey)
 		keys = append(keys, keyV2)
-		state := ethkey.State{
-			Address:    keyV1.Address,
-			NextNonce:  keyV1.NextNonce,
-			IsFunding:  keyV1.IsFunding,
-			EVMChainID: *utils.NewBig(chainID),
-		}
-		states = append(states, state)
+		nonces = append(nonces, keyV1.NextNonce)
+		fundings = append(fundings, keyV1.IsFunding)
+	}
+	return keys, nonces, fundings, nil
+}
+
+// XXXTestingOnlySetState is only used in tests to manually update a key's state
+func (ks *eth) XXXTestingOnlySetState(state ethkey.State) {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		panic(ErrLocked)
+	}
+	existingState, exists := ks.keyStates.ChainIDKeyID[state.EVMChainID.String()][state.KeyID()]
+	if !exists {
+		panic(fmt.Sprintf("key not found with ID %s", state.KeyID()))
+	}
+	*existingState = state
+	sql := `UPDATE evm_key_states SET address = :address, next_nonce = :next_nonce, is_disabled = :is_disabled, evm_chain_id = :evm_chain_id, updated_at = NOW()
+	WHERE address = :address;`
+	_, err := ks.orm.q.NamedExec(sql, state)
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+/// XXXTestingOnlyAdd is only used in tests to manually add a key
```

Comment:
Confusingly, it was this triple slash: :roll_eyes: 
```suggestion
// XXXTestingOnlyAdd is only used in tests to manually add a key
```

Reply:
lmao